indico/modules/events/contributions/controllers/display.py

```diff
@@ -0,0 +1,165 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from io import BytesIO
+
+from flask import session, request, jsonify
+from sqlalchemy.orm import load_only, noload, joinedload
+
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.persons import ContributionPersonLink, AuthorType
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.contributions.util import (get_contributions_with_user_as_submitter,
+                                                      serialize_contribution, ContributionDisplayReporter)
+from indico.modules.events.contributions.views import WPMyContributions, WPContributions
+from indico.modules.events.util import get_base_ical_parameters
+from indico.web.flask.util import send_file, jsonify_data
+from indico.web.http_api.metadata.serializer import Serializer
+from MaKaC.PDFinterface.conference import ContribToPDF, ContribsToPDF
+from MaKaC.webinterface.rh.base import RH
+from MaKaC.webinterface.rh.conferenceDisplay import RHConferenceBaseDisplay
+
+
+class RHContributionDisplayBase(RHConferenceBaseDisplay):
+
+    CSRF_ENABLED = True
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib
+        }
+    }
+
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.contrib = Contribution.get_one(request.view_args['contrib_id'], is_deleted=False)
```

Comment:
I was gonna say `is_deleted` is not needed, but just noticed we're not retrieving it through the event-contained relationship.